src/ng/controller.js

```diff
@@ -3,7 +3,7 @@
 var $controllerMinErr = minErr('$controller');
 
 
-var CNTRL_REG = /^(\S+)(\s+as\s+(\w+))?$/;
+var CNTRL_REG = /^(\S+)(\s+as\s+([\w$]+))?$/;
```

Comment:
Yeah, it crossed my mind, but I thought it was too strict. It complicates things unnecessarily, I think.
(Note that it's already possible to use many more characters using `controllerAs`.)